### PR TITLE
Updated copy of #12

### LINKED ISSUE #8: Blacklist Apps
For 99% of the apps I want to allow them to steal my focus, and always pop-up, but for some apps I want to get a pop-up message instead.

most importantly for Microsoft Teams, their notification system bypasses dbus, and thus steals focus whenever a notification pops up.
Which means every 15 minutes my typing stops responding because I got a notification for a message that stole my focus.

### PR DESCRIPTION
**Pull request by balacij**
Closes #8 
Builds on #12 (increases Gnome version minimum requirement to 40+, and is the majority of this PR)

Continues @skoruppa's work with the feedback from @bdaase. Please let me know if you need any other changes.

### CODE REVIEW

**Comment by sudo-chan** [comment]
I just tried installing your fork and I got this error on the extensions app. 

OS: Pop OS 20.04
Gnome: 3.36.8

---

The settings of extension noannoyance@daase.net had an error:
```
TypeError: enableIgnorelistBox.append is not a function
```

Stack trace:
```
buildPrefsWidget@/home/sharan/.local/share/gnome-shell/extensions/noannoyance@daase.net/prefs.js:51:25
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

This is how I installed it:

```
git clone https://github.com/balacij/noannoyance.git
mv noannoyance ~/.local/share/gnome-shell/extensions/noannoyance@daase.net
```

**Comment by balacij** [comment]
Hi @sudo-chan, I _think_ the issue is a version mismatch. It looks like @skoruppa's work had the minimum Gnome version raised to 40+. I can probably ask around if anyone knows an easy way to resolve this kind of issue, but this will take time. If you're interested in immediately using this, however, you will need to upgrade to Gnome 40+ (I imagine upgrading to [Pop OS 22.04](https://blog.system76.com/post/682519660741148672/popos-2204-lts-has-landed) would work for you).

**Comment by skoruppa** [comment]
@balacij thanks! Had no time to make changes based on @bdaase feedback 

@sudo-chan It is exactly as @balacij stated. Most extensions I have found have different versions for the current and previous Gnome releases (because of the migration from GTK3 to GTK4). Maybe there is a way to write an extension that would be compatible with both but I think I'm too stupid to do it ;) 

If you want to use the whitelist functionality with 3.36.8 use my [initial commit ](https://github.com/bdaase/noannoyance/pull/12/commits/94e16343964d9b6f5d29b972703139eec2054ac4) from #12 (it was not merged because it was not compatible with GTK4 ;) 

Maybe @bdaase could use my initial commit to make an extension version for Gnome up to 40, and changes from here for a version for Gnome 40+

File: prefs.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ function buildPrefsWidget() {
     });
 
     const indicatorIdColumn = new Gtk.TreeViewColumn({
-        title: "WM__CLASS List",
+        title: 'WM_CLASS List ("Alt + F2" > Run "lg" > Click "Windows")',
```
Comment (bdaase) [inline prefs.js:78]: Here the escape of the `_` got lost.

**Comment by bdaase** [comment]
@balacij Thanks for picking that up. Just one tiny comment.
Besides that, please agree with @skoruppa on authorship of this MR.

> Maybe @bdaase could use my initial commit to make an extension version for Gnome up to 40, and changes from here for a version for Gnome 40+

I don't think this will happen. I have super little time at the moment and don't see the point in maintaining two different versions of the extension, of which one is just for compatibility with over two years old, discontinued by upstream, software.

File: prefs.js (hunk 1 above)
Comment (balacij) [reply prefs.js]: Fixed, thanks!

**Comment by balacij** [comment]
@skoruppa Cheers! Thank you as well for your work on it! Glad I could help out a bit :smile: 

@bdaase Thanks! I don't really care about having authorship of the PR because the git log will still show authorship appropriately by commits. If the GitHub PR authorship matters to anyone, we should have @skoruppa as the author. In which case, he will need to merge my PR to his fork (https://github.com/skoruppa/noannoyance/pull/1) and then we can close this one. Alternatively, however, we can merge both PRs, unlinking mine from closing the original ticket.

**Comment by bdaase** [comment]
@balacij I will squash and merge them, as the individual commits don't really add much value. And indeed, this would have you as the author and @skoruppa as the co-author. If this is fine with both of you, I would go for that.

> This is also a really great extension, and this feature might be noteworthy for the [Gnome Weekly posts](https://thisweek.gnome.org/).

Thanks! Once merged, feel free to let them know about the new feature :)

**Comment by balacij** [comment]
> @balacij I will squash and merge them, as the individual commits don't really add much value. And indeed, this would have you as the author and @skoruppa as the co-author. If this is fine with both of you, I would go for that.

Hmm, in that case, it would be best to have @skoruppa as the main author (and me as the co-author).

> > This is also a really great extension, and this feature might be noteworthy for the [Gnome Weekly posts](https://thisweek.gnome.org/).
> 
> Thanks! Once merged, feel free to let them know about the new feature :)

Will do!

**Comment by bdaase** [comment]
> Hmm, in that case, it would be best to have @skoruppa as the main author (and me as the co-author).

Alright, but I don't think this is possible when merging this PR. You would somehow have to merge your changes into the original PR, and we would merge that.

> Will do!

Awesome, thanks!

**Comment by balacij** [comment]
> Alright, but I don't think this is possible when merging this PR. You would somehow have to merge your changes into the original PR, and we would merge that.

Sounds good. I'll close this PR, and we can wait for @skoruppa to pull in my changes from his fork. His PR should then be updated, and you can merge his.

**Pull request closed without merging**